channeldb/payment_control.go

```diff
@@ -720,21 +720,16 @@ func (p *PaymentControl) FetchInFlightPayments() ([]*MPPayment, error) {
 				return fmt.Errorf("non bucket element")
 			}
 
-			// If the status is not InFlight, we can return early.
-			paymentStatus, err := fetchPaymentStatus(bucket)
+			p, err := fetchPayment(bucket)
 			if err != nil {
 				return err
 			}
 
-			if paymentStatus != StatusInFlight {
+			// Skip the payment if it's terminated.
+			if p.Terminated() {
```

Comment:
If this is a 100% refactor without a functional change, it would be good to note that in the commit msg. It isn't always clear to a reviewer what to pay special attention to in a commit. If you know it is supposed to be exactly equivalent, it is easier to verify.

Reply:
Updated the commit message.